### PR TITLE
Keep get tables API with and without database

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
From cluster management purpose, we need APIs to read table names from all databases. Keep the original APIs the same (without filtering on database), and for database aware purpose, use the new APIs.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -821,7 +804,7 @@ public List<String> getAllRealtimeTables(@Nullable String databaseName) {
    * @return List of raw table names in default database
```
Comment (zhtaoxiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:804]: `in default database` -> `in all database`

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 2]
```diff
@@ -831,14 +814,9 @@ public List<String> getAllRawTables() {
...
-    for (String resourceName : getAllResources()) {
-      if (TableNameBuilder.isTableResource(resourceName)
-          && DatabaseUtils.isPartOfDatabase(resourceName, databaseName)) {
-        rawTableNames.add(TableNameBuilder.extractRawTableName(resourceName));
-      }
-    }
-    return new ArrayList<>(rawTableNames);
+    return getAllResources().stream().filter(
+        resourceName -> TableNameBuilder.isTableResource(resourceName) && DatabaseUtils.isPartOfDatabase(resourceName,
+            databaseName)).distinct().collect(Collectors.toList());
```
Comment (zhtaoxiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: nit: could you please help me understand why do we need to call `distinct()`?

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 3]
```diff
@@ -831,14 +815,9 @@ public List<String> getAllRawTables() {
...
-    for (String resourceName : getAllResources()) {
-      if (TableNameBuilder.isTableResource(resourceName)
-          && DatabaseUtils.isPartOfDatabase(resourceName, databaseName)) {
-        rawTableNames.add(TableNameBuilder.extractRawTableName(resourceName));
-      }
-    }
-    return new ArrayList<>(rawTableNames);
+    return getAllResources().stream().filter(
+        resourceName -> TableNameBuilder.isTableResource(resourceName) && DatabaseUtils.isPartOfDatabase(resourceName,
+            databaseName)).map(TableNameBuilder::extractRawTableName).distinct().collect(Collectors.toList());
```
Comment (zhtaoxiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:820]: nit: could you please help me understand why we need to call `distinct()`?

Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:820]: E.g. `myTable_OFFLINE` and `myTable_REALTIME` should result in one raw table name `myTable`

**Review by zhtaoxiang — APPROVED** [review]
LGTM

**Review by xiangfu0 — APPROVED** [review]

**Pull request merged**